examples/file-config-solver/file-config-solver.cpp

```diff
@@ -0,0 +1,150 @@
+    auto host_b = vec::create(exec->get_master(), gko::dim<2>(size, 1));
+    for (auto i = 0; i < size; i++) {
+        host_x->at(i, 0) = 0.;
+        host_b->at(i, 0) = 1.;
+    }
+    auto x = vec::create(exec);
+    auto b = vec::create(exec);
+    x->copy_from(host_x);
+    b->copy_from(host_b);
+
+    // Calculate initial residual by overwriting b
+    auto one = gko::initialize<vec>({1.0}, exec);
+    auto neg_one = gko::initialize<vec>({-1.0}, exec);
+    auto initres = gko::initialize<vec>({0.0}, exec);
+    A->apply(one, x, neg_one, b);
+    b->compute_norm2(initres);
+
+    // Copy b again
+    b->copy_from(host_b);
+
+    // Read the json file into ginkgo structure
+    // cg.json: simple-solver
+    // blockjacobi-cg.json: preconditioned-solver
+    // ir.json: iterative-refinement
+    // parilu.json: ilu-preconditioned-solver (by using factoization parameter
+    //              directly)
+    // pgm-multigrid-cg.json: multigrid-preconditioned-solver (set
+    //                        min_coarse_rows additionally due to this small
+    //                        example matrix)
+    // mixed-pgm-multigrid-cg.json: mixed-multigrid-preconditioned-solver
+    //                              (assuming there are always more than one
+    //                              level)
+    auto config = gko::ext::config::parse_json_file(configfile);
+    // Create the registry, which allows passing the existing data into config
+    // This example does not show the usage
+    auto reg = gko::config::registry();
+    // Create the default type descriptor, which gives the default common type
+    // (value/index) for solver generation. If the solver does not specify value
+    // type, the solver will use these types.
+    auto td = gko::config::make_type_descriptor<ValueType, IndexType>();
+    // generate the linopfactory on the given executors
+    auto solver_gen = gko::config::parse(config, reg, td).on(exec);
+
+    // Create solver
+    std::chrono::nanoseconds gen_time(0);
+    auto gen_tic = std::chrono::steady_clock::now();
+    auto solver = solver_gen->generate(A);
+    exec->synchronize();
+    auto gen_toc = std::chrono::steady_clock::now();
+    gen_time +=
+        std::chrono::duration_cast<std::chrono::nanoseconds>(gen_toc - gen_tic);
+
+    // Add logger
+    std::shared_ptr<const gko::log::Convergence<ValueType>> logger =
+        gko::log::Convergence<ValueType>::create();
+    solver->add_logger(logger);
+
+    // Solve system
+    exec->synchronize();
+    std::chrono::nanoseconds time(0);
+    auto tic = std::chrono::steady_clock::now();
+    solver->apply(b, x);
+    exec->synchronize();
+    auto toc = std::chrono::steady_clock::now();
+    time += std::chrono::duration_cast<std::chrono::nanoseconds>(toc - tic);
+
+    // Print out the solver config
+    std::cout << "Config file: " << configfile << std::endl;
+    std::ifstream f(configfile);
+    std::cout << f.rdbuf() << std::endl;
```

Comment:
You could output the json directly:
```
std::cout << std::setw(4) << nlohman::json::parse(configfile) << std::endl;
```
see: https://json.nlohmann.me/api/operator_ltlt/
Of course, it would make more sense if the json object was stored already before.

Reply:
Thanks for this information.
I will put it when we have the example with loading json object before